code/Test_Definitions/NumberVerification_verify.feature

```diff
@@ -0,0 +1,137 @@
+
+
+@NumberVerification_verify
+Feature: Camara Number Verification API verify
+
+# Input to be provided by the implementation to the tests
+# References to OAS spec schemas refer to schemas specified in
+# https://raw.githubusercontent.com/camaraproject/NumberVerification/main/code/API_definitions/number_verification.yaml
+
+  Background: Common Number Verification verify setup
+    Given the resource "/number-verification/v0"  as  base url
+    And the header "Content-Type" is set to "application/json"
+    And the header "Authorization" is set to a valid access token
+    And the header "x-correlator" is set to a UUID value
+    And the request body is compliant with the schema NumberVerificationRequestBody
+    And the response body is compliant with the schema NumberVerificationMatchResponse
+    And the header "x-correlator" is set to a UUID value
+    And NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1 is compliant with the schema DevicePhoneNumber
+    And NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER2 is compliant with the schema DevicePhoneNumber
+    And NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER1 is different to NUMBERVERIFY_VERIFY_MATCH_PHONENUMBER2
+
+  @NumberVerification_verify0_phoneNumber_does_not_match_schema
+  Scenario Outline: phoneNumber value does not comply with the schema
+    Given the request body property "$.phoneNumber" is set to: <phone_number_value>
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
+      | phone_number_value |
+      | string_value       |
+      | 1234567890         |
+      | +12334foo22222     |
+      | +00012230304913849 |
+      | 123                |
+      | ++49565456787      |
+
+  @NumberVerification_verify1_xcorrelator_does_not_match_schema
+  Scenario Outline: x-correlator request header value does not comply with the schema
+    Given if the optional request header "x-correlator" is set to: <xcorrelator_value>
+    When the HTTP "POST" request is sent
+    Then the response status code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+    Examples:
```

Comment:
These examples look like valid x-correlator values to me (`string_value` and `boink`) that do comply with the schema. 

The [x-correlator schema](https://github.com/camaraproject/NumberVerification/blob/main/code/API_definitions/number_verification.yaml#L87) states that the value should be a string (no other restrictions):
```
      parameters:
        - in: header
          name: X-Correlator
          required: false
          description: Correlation id for the different services
          schema:
            type: string
```

Not sure how we would pass in an invalid value (e.g. passing in a number instead of string). I just tested in Postman and tried to pass in a number, but the generated cURL does not differentiate between string and number header values:

```
curl --location '/number-verification/v0/verify' \
--header 'Authorization: Bearer abc' \
--header 'Content-Type: application/json' \
--header 'accept: application/json' \
--header 'x-correlator: 123' \
--data '{
  "phoneNumber": "+4412345678"
}'
```

Reply:
Same view that Toyeeb - not sure if we are able to test when x-correlator did not match the schema end if we need in this case to send a 400.  We are not so strict in https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#9-architecture-headers

Reply:
My bad. Maybe I took the x-correlator UUID examples as binding. I thought that x-correlator must be a UUID.
Actually I think that would be a useful requirement because it makes it harder to misuse the header x-correlator.
The value is comming "from the outside" and security usually recommends input validation.
Currently the value of x-correlator could be 64K times the letter 'A' or "Robert'); DROP TABLE Users;--".

"String" is hard to validate in e.g. an API Firewall.

I'll remove that test.

Is there a way to specify: For all tests that return a status of 200, check that the value of x-correlator in the request is the same as the value in the response?